src/rdf4cpp/BigDecimal.hpp

```diff
@@ -727,18 +703,20 @@ struct BigDecimal {
             return this->positive() ? std::strong_ordering::greater : std::strong_ordering::less;
         UnscaledValue_t t = this->unscaled_value;
         UnscaledValue_t o = other.unscaled_value;
-        if (this->exponent > other.exponent) {
-            UnscaledValue_t b{base};
-            if (pow_checked<OverflowMode::Checked>(b, this->exponent - other.exponent, b))
-                return std::strong_ordering::less;  // t does fit into the same precision, while o does not
-            if (mul_checked<OverflowMode::Checked>(o, b, o))
-                return std::strong_ordering::less;  // t does fit into the same precision, while o does not
-        } else if (this->exponent < other.exponent) {
-            UnscaledValue_t b{base};
-            if (pow_checked<OverflowMode::Checked>(b, other.exponent - this->exponent, b))
-                return std::strong_ordering::greater;  // o does fit into the same precision, while t does not
-            if (mul_checked<OverflowMode::Checked>(t, b, t))
-                return std::strong_ordering::greater;  // o does fit into the same precision, while t does not
+        if (t != 0 && o != 0) {  // if one of the decimals is 0, compare directly (0*base^e is 0 for all e)
```

Comment:
the comment does not match the logic. The condition just says that either must not be null, both could be non-null